### PR TITLE
New documentation site

### PR DESCRIPTION
**Pull request by phil-scale**
# Pull Request Summary
Create documentation site generated from markdown files. Uses mkdocs and some extensions for nice formatting.

# Test Plan and Usage Guide
https://scaleapi.github.io/launch-python-client/

Shortcut Ticket
[[sc-719880]](https://app.shortcut.com/scaleai/story/719880)

### CODE REVIEW

File: docs/concepts/callbacks.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+client = LaunchClient(api_key=os.getenv("LAUNCH_API_KEY"))
+endpoint = client.create_model_endpoint(
+    endpoint_name="demo-endpoint-callback",
+    model_bundle="test-bundle",
+    cpus=1,
+    min_workers=1,
+    endpoint_type="async",
+    update_if_exists=True,
+    labels={
+        "team": "infra",
```
Comment (yixu34) [inline docs/concepts/callbacks.md:37]: 👀

Comment (yixu34) [reply docs/concepts/callbacks.md:37]: Would feel more comfortable with a link to our canonical team list.

File: docs/concepts/model_endpoints.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+of CPUs, amount of memory, GPU count, and type of GPU.
+
+Endpoints can be asynchronous or synchronous. Asynchronous endpoints return
+a future immediately after receiving a request, and the future can be used to
+retrieve the prediction once it is ready. Synchronous endpoints return the
+prediction directly after receiving a request.
+
+## Creating Async Model Endpoints
+
+Async model endpoints are the most cost-efficient way to perform inference on
```
Comment (yixu34) [inline docs/concepts/model_endpoints.md:17]: While we're here, might be worth doing a small content refresh. My thinking is that we can lift this paragraph and the corresponding one under `Creating Sync Endpoints` and centralize them into the top-level `Model Endpoints` section, or have a dedicated sub-section explaining the tradeoffs.

Would also suggest adding a note about the tradeoffs w.r.t responsibilities of queue management to supplement the existing highlighted tradeoff between latency and cost.

File: docs/concepts/callbacks.md (hunk 1 above)
Comment (yixu34) [reply docs/concepts/callbacks.md:37]: Oh wait this is a public repo. Hmm...

Do we still require the notion of teams for external?

**Review by yixu34 — COMMENTED** [review]
Main concern is around defaulting to `infra` in the docs.

Also would be good to tag some people outside of the team for documentation-related PRs to give a heads up, especially if they're going to be collaborating or consuming this stuff very closely. cc @adlam-scale @harish-kamath @saiatmakuri @chongzluong

File: docs/concepts/callbacks.md (hunk 1 above)
Comment (phil-scale) [reply docs/concepts/callbacks.md:37]: Created https://app.shortcut.com/scaleai/story/727036 to track

File: docs/concepts/model_endpoints.md (hunk 2 above)
Comment (phil-scale) [reply docs/concepts/model_endpoints.md:38]: image

File: tests/test_docs.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    module_name,
+    source_code,
+    import_execute,
+    mocker,
+    mock_dictionary,
+    mock_model_bundle,
+    mock_async_endpoint,
+    mock_batch_job,
+):
+    mocker.patch("launch.connection.Connection", MagicMock())
```
Comment (yixu34) [inline tests/test_docs.py:106]: IIRC, we're getting rid of this `Connection` Nucleus-ism after the client overhaul?

**Review by yixu34 — APPROVED** [review]

File: tests/test_docs.py (hunk 3 above)
Comment (phil-scale) [reply tests/test_docs.py:106]: 😢 unfortunately there's still some routes, like bundle upload, that aren't in our OpenAPI schema... I guess we could add them and then we could remove the `Connection` Nucleus-ism

**Pull request merged**